src/com/sap/piper/versioning/DockerArtifactVersioning.groovy

```diff
@@ -0,0 +1,50 @@
+package com.sap.piper.versioning
+
+class DockerArtifactVersioning extends ArtifactVersioning {
+    protected DockerArtifactVersioning(script, configuration) {
+        super(script, configuration)
+    }
+
+    @Override
+    def getVersion() {
+        if (configuration.dockerVersionSource == 'FROM')
+            return  getVersionFromDockerBaseImageTag(configuration.filePath)
+        else
+            //standard assumption: version is assigned to an env variable
+            return getVersionFromDockerEnvVariable(configuration.filePath, configuration.dockerVersionSource)
+    }
+
+    @Override
+    def setVersion(version) {
+        def dockerVersionDir = (configuration.dockerVersionDir?dockerVersionDir:'')
+        script.dir(dockerVersionDir) {
+            script.writeFile file:'VERSION', text: version
+        }
+    }
+
+    def getVersionFromDockerEnvVariable(filePath, name) {
+        def lines = script.readFile(filePath).split('\n')
+        def version = ''
+        for (def i = 0; i < lines.size(); i++) {
+            if (lines[i].startsWith('ENV') && lines[i].split(' ')[1] == name) {
```

Comment:
It would be nice, if we also consider `ARG` variable.

Reply:
I was considering ARG but the issue with it is that ARG can be overwritten during Docker build. I would therefore rather leave it out for now.